task/backend/coordinator.go

```diff
@@ -0,0 +1,43 @@
+package backend
+
+import (
+	"context"
+
+	"github.com/influxdata/influxdb"
+)
+
+// Tasksfinder is a type on which tasks can be listed
+type TasksFinder interface {
+	FindTasks(context.Context, influxdb.TaskFilter) ([]*influxdb.Task, int, error)
+}
+
+// Coordinator is a type with a single method which
+// is called when a task has been created
+type Coordinator interface {
+	TaskCreated(context.Context, *influxdb.Task) error
+}
+
+// NotifyCoordinatorOfCreated lists all tasks by the provided task service and for
+// each task it calls the provided coordinators task created method
+func NotifyCoordinatorOfCreated(ctx context.Context, ts TasksFinder, coord Coordinator) error {
+	// If we missed a Create Action
+	tasks, _, err := ts.FindTasks(ctx, influxdb.TaskFilter{})
+	if err != nil {
+		return err
+	}
+
+	for len(tasks) > 0 {
+		for _, task := range tasks {
+			coord.TaskCreated(ctx, task)
```

Comment:
One thing that we used to do is if influxdb had been shot down for any amount of time we would not want the oss version to do a "catch up" for the time the task's were down. I am not sure if that code was moved somewhere else but I suspect we will still want to update the existing tasks with a latestCompleted time to time.Now.

Reply:
oooh good catch